File: pubsub/src/main/java/glitch/pubsub/object/json/SubscriptionMessage.java
package glitch.pubsub.object.json;

import com.google.gson.annotations.SerializedName;
import glitch.api.objects.enums.SubscriptionType;
import glitch.pubsub.object.enums.SubscriptionContext;
import lombok.Data;

import java.time.Instant;

@Data
public class SubscriptionMessage {
    @SerializedName("user_name")
    private final String username;
    private final String displayName;
    private final String channelName;
    private final Long userId;
    private final Long channelId;
    private final Instant time;
    @SerializedName("sub_plan")
    private final SubscriptionType subscriptionType;
    @SerializedName("sub_plan_name")
    private final String subscriptionName;
    private final Integer months;
    private final SubscriptionContext context;
    @SerializedName("sub_message")
    private final OrdinalMessage message;
}

File: pubsub/src/main/kotlin/glitch/pubsub/object/json/SubscriptionMessage.kt
package glitch.pubsub.`object`.json

import com.google.gson.annotations.SerializedName
import glitch.api.objects.enums.SubscriptionType
import glitch.pubsub.`object`.enums.SubscriptionContext
import java.time.Instant

open class SubscriptionMessage(
        @SerializedName("user_name")
        val username: String, val displayName: String,
        val channelName: String, val userId: Long,
        val channelId: Long, val time: Instant,
        @SerializedName("sub_plan")
        val subscriptionType: SubscriptionType,
        @SerializedName("sub_plan_name")
        val subscriptionName: String?,
        val months: Int, val context: SubscriptionContext,
        @SerializedName("sub_message")
        val message: OrdinalMessage?) {

}
